src/util.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  GetObjectCommand,
+  GetObjectCommandOutput,
+  S3Client,
+} from "@aws-sdk/client-s3";
+import { exception } from "./errors";
+import { localFromISO } from "./datetime";
+import { OCSEvent } from "./processor/structs";
+import { create as struct } from "superstruct";
+
+export const safeSend = async (
+  client: S3Client,
+  bucket: string,
+  key?: string
+): Promise<GetObjectCommandOutput> => {
+  const command = new GetObjectCommand({ Bucket: bucket, Key: key });
+  const response = await client.send(command);
+  const { $metadata: metadata, Body: data } = response;
+
+  if (data === undefined)
+    throw exception("BadS3Response", JSON.stringify(metadata));
+
+  return response;
+};
+
+export const recoverLine = (line: string): string => {
+  const { rawData } = JSON.parse(line);
+  const eventData = JSON.parse(Buffer.from(rawData, "base64").toString());
```

Comment:
This is expecting that `eventData` is always an object with a single event. Once https://github.com/mbta/trike/pull/20 lands, this will no longer be the case: it could be a object or a list of objects. #238 supports this use case for the non-recovery case, and the same pattern (wrap the event in a list, return a string with 1 or more lines) should be usable here as well.